mvc/controllers/actions.rst

```diff
@@ -1,8 +1,63 @@
-.. include:: /../common/stub-topic.txt
+Actions and Action Results
+==========================
 
-|stub-icon| Actions and Action Results
-===========================================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+Actions and action results are a fundamental part of how developers build apps using ASP.NET MVC.
 
-.. _issue: https://github.com/aspnet/Docs/issues/118
+Sections:
+	- `What is a Controller`_
+	- `Defining Actions`_
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/1.0.0-rc1/mvc/performance/response-caching/sample>`_.
+
+What is a Controller
+--------------------
+
+In ASP.NET MVC, a `Controller` is used to define and group a set of actions. An `action` (or `action method`) is a method on a controller that handles incoming requests. In ASP.NET 5, a controller can be any instantiable class that ends in "Controller" or inherits from a class that ends with "Controller". By convention, controller classes may be placed within a root-level "Controllers" folder within the app, and may inherit from ``Microsoft.AspNet.Mvc.Controller``, but these are not required.
+
+Controllers support :doc:`dependency injection <dependency-injection>`, and should follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle>`_ and request any dependencies their actions require through their constructor. Controllers provide a logical means of grouping similar actions together, allowing common sets of rules (e.g. routing, caching, authorization) to be applied collectively.
```

Comment:
We're covering controllers in this article since it's the first one in the Controllers section of the TOC and there isn't (currently) a separate controllers article discussing them at a high level like this. I've addressed the above feedback in the next commit I'll be making.